.github/workflows/on-pull.yml

```
@@ -0,0 +1,25 @@
+name: Run Tests and Linting
+
+on:
+  pull_request:
+    branches:
+      - master
```

You might want `on: push` here. Otherwise, if you make changes to a PR after the initial push, the tests will not run. (If so, change the file name too.)

Also, to optimize check suite run time in these monorepo projects, I've been adding the `paths` property which restricts an action to running only when there are changes under the specified path.

```suggestion
on:
  push:
    paths:
      - 'packages/analytics-adapter/**'
```

---

Done